cupy/cusparse.py

```diff
@@ -1605,6 +1606,10 @@ def csrsm2(a, b, alpha=1.0, lower=True, unit_diag=False, transa=False,
           a_desc.descriptor, a.data.data.ptr, a.indptr.data.ptr,
           a.indices.data.ptr, b.data.ptr, ldb, info, policy, ws.data.ptr)
 
+    # without sync we'd get either segfault or cuda context error
+    _stream.get_current_stream().synchronize()
+    _cusparse.destroyCsrsm2Info(info)
```

Comment:
These objects are not reusable right?

Reply:
You meant `info`? It'd be reusable as mentioned in the PR description, but it'd take another PR to rewrite the whole  thing to make it useful. Let's get the fix in first, since this function [will be removed in CUDA 12](https://docs.nvidia.com/cuda/cusparse/index.html#csrsm2_solve) it might not worth the effort (not sure).

Reply:
yeah, SGTM